### PR TITLE
Started working on integration tests.

### PR DESCRIPTION
**Pull request by j-bennet**
Copy of PR #266.

### CODE REVIEW

**Comment by amjith** [comment]
@j-bennet Is there more to be done here or should we merge this in and start adding more behavioral tests?

**Comment by j-bennet** [comment]
I think it' good to merge. I'll add more tests later, but the current setup can also be used by others to build upon.

**Comment by amjith** [comment]
Would you mind adding some instructions to Readme or Develop.rst on how to run this?

I tried running `behave` and I get the following error:

```
Exception OperationalError: FATAL:  role "behave_user" does not exist

Traceback (most recent call last):
  File "/Users/amjith/.virtualenvs/pg-cli/bin/behave", line 11, in <module>
    sys.exit(main())
  File "/Users/amjith/.virtualenvs/pg-cli/lib/python2.7/site-packages/behave/__main__.py", line 109, in main
    failed = runner.run()
  File "/Users/amjith/.virtualenvs/pg-cli/lib/python2.7/site-packages/behave/runner.py", line 672, in run
    return self.run_with_paths()
  File "/Users/amjith/.virtualenvs/pg-cli/lib/python2.7/site-packages/behave/runner.py", line 693, in run_with_paths
    return self.run_model()
  File "/Users/amjith/.virtualenvs/pg-cli/lib/python2.7/site-packages/behave/runner.py", line 471, in run_model
    self.run_hook('before_all', context)
  File "/Users/amjith/.virtualenvs/pg-cli/lib/python2.7/site-packages/behave/runner.py", line 405, in run_hook
    self.hooks[name](context, *args)
  File "features/environment.py", line 48, in before_all
    context.conf['dbname'])
  File "/Users/amjith/Dropbox/code/python/pgcli/features/db_utils.py", line 19, in create_db
    cn = create_cn(hostname, password, username, 'postgres')
  File "/Users/amjith/Dropbox/code/python/pgcli/features/db_utils.py", line 47, in create_cn
    cn = connect(host=hostname, user=username, database=dbname)
  File "/Users/amjith/.virtualenvs/pg-cli/lib/python2.7/site-packages/psycopg2/__init__.py", line 164, in connect
    conn = _connect(dsn, connection_factory=connection_factory, async=async)
psycopg2.OperationalError: FATAL:  role "behave_user" does not exist
```

Could you create the user (or use `postgres` user) as part of the test? 

Also create the database that is needed to run these tests? 

Eg: We do something similar in our regular test suite: https://github.com/dbcli/pgcli/blob/master/tests/utils.py

**Comment by j-bennet** [comment]
Will do.

**Comment by j-bennet** [comment]
Added instructions to DEVELOP.rst (looks like a better place than normal readme?). Changed default test user to postgres.

**Comment by amjith** [comment]
Were you able to get the tests passing? The \? test always times out for me.

**Comment by j-bennet** [comment]
Yes. Timing out means something's wrong though. What's the trace?

**Comment by j-bennet** [comment]
If you look here:

https://github.com/dbcli/pgcli/blob/j-bennet/itegration-tests/features/fixture_data/help.txt

This is what the test expects to be output when user enters `\?` command. So what can happen is, maybe your output is somehow different, and then the process waits and waits for this expected output and never gets it, until it times out.

What do you see in your cli when you enter `\?`?

**Comment by amjith** [comment]
I was using 'fancy_grid' for the table_format in .pgclirc. 

So the output didn't match exactly. Can we use a specific table_format in the integration tests?

**Comment by j-bennet** [comment]
Ah, that explains it. I should set all the output parameters explicitly then, they can be set when calling pgcli from command line, correct?

**Comment by amjith** [comment]
I don't think there is a way to set the grid at launch time via command line arg. It relies on the config file.

**Comment by j-bennet** [comment]
Is there a way to start with a given configuration file?

**Comment by amjith** [comment]
I was thinking the same thing. Not right now but that's a good idea. We should add an option to pass in the config file.

**Comment by j-bennet** [comment]
Ok, so I'm not expecting the exact table output anymore. The tests should pass now. But definitely it would be nice to add the config option. Makes reproducing the exact environment so much easier. Do you mind if I do that (in a different branch)?

File: DEVELOP.rst [hunk 1]
```diff
@@ -122,3 +122,33 @@ The rpm package can be installed as follows:
...
     $ sudo yum install pgcli*.rpm
+
+Running the integration tests
+-----------------------------
+
+Integration tests use `behave package http://pythonhosted.org/behave/`_.
+Configuration settings for this package are provided via ``behave.ini`` file
+in root directory.
+
+The database user ``pg_test_user`` has to have permissions to create and drop
```
Comment (amjith) [inline DEVELOP.rst]: This should be `postgres` user right?

**Comment by amjith** [comment]
After you removed the formatting, it passes the tests as expected. 

Thanks for taking the time to write integrations tests for pgcli. I appreciate it very much. :)

File: DEVELOP.rst (hunk 1 above)
Comment (j-bennet) [inline DEVELOP.rst]: The setting in .ini file is called "pg_test_user". I updated the readme to make that clear.

**Comment by amjith** [comment]
Thank you! Looks great.

**Pull request merged**